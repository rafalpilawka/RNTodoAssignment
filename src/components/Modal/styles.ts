import {StyleSheet} from 'react-native';
import {colors} from 'src/constants/platformSpecificColors';
import normalize from 'react-native-normalize'

export const styles = StyleSheet.create({
  firstContainer: {
    backgroundColor: 'rgba(0,0,0,0.5)',
    flex: 1,
    justifyContent: 'flex-end',
  },
  safeArea: {
    justifyContent: 'flex-end',
    backgroundColor: colors!.color,
  },
  secondContainer: {
    backgroundColor: colors!.color,
    paddingHorizontal: '5%',
    alignItems: 'center',
    justifyContent: 'flex-start',
    paddingBottom: normalize(25),
    height: normalize(200),
  },
  itemsContainer: {
    alignItems: 'flex-start',
    width: '100%',
  },
});
