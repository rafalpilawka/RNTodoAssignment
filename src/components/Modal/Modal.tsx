import React, {ReactElement} from 'react';
import {
  Modal as RNModal,
  SafeAreaView,
  TouchableOpacity,
  KeyboardAvoidingView,
  Platform,
} from 'react-native';
import {styles} from './styles';

interface Props {
  visible: boolean;
  onClose?: () => void;
  children?: ReactElement;
  onDismiss?(): void;
}

export const Modal = ({visible, children, onDismiss}: Props): ReactElement => {
  const _render = () => {
    return (
      <TouchableOpacity
        testID={'dismissPlane'}
        style={styles.firstContainer}
        onPress={onDismiss}>
        <KeyboardAvoidingView
          behavior={Platform.OS === 'ios' ? 'padding' : 'height'}>
          <SafeAreaView style={[styles.safeArea]}>
            <TouchableOpacity
              style={[styles.secondContainer]}
              onPress={(e) => e.preventDefault()}>
              {children}
            </TouchableOpacity>
          </SafeAreaView>
        </KeyboardAvoidingView>
      </TouchableOpacity>
    );
  };

  return (
    <RNModal
      animationType={'slide'}
      supportedOrientations={['portrait']}
      visible={visible}
      transparent
      onRequestClose={onDismiss}>
      {_render()}
    </RNModal>
  );
};
