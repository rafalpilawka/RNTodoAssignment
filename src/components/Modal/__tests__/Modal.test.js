import React from 'react';
import {shallow} from 'enzyme';
import {Text} from 'react-native';
import {Modal} from 'src/components/Modal/Modal';
import renderer from 'react-test-renderer';

const props = {
  visible: true,
  onClose: jest.fn(),
  onDismiss: jest.fn(),
};

describe('Modal', () => {
  let component;
  const testLabel = 'test123';
  beforeEach(() => {
    component = shallow(
      <Modal {...props}>
        {' '}
        <Text>{testLabel}</Text>
      </Modal>,
    );
    jest.clearAllMocks();
  });
  it('should match to snapshot', () => {
    const component = renderer.create(<Modal {...props} />).toJSON();
    expect(component).toMatchSnapshot();
  });

  it('should be visible', () => {
    const modalVisibleProp = component.dive().props().visible;
    expect(modalVisibleProp).toEqual(props.visible);
  });
  it('should not be visible', () => {
    component = shallow(<Modal {...props} visible={false} />);
    const modalVisibleProp = component.dive().props().visible;
    expect(modalVisibleProp).toBeFalsy();
  });
  it('should close modal', () => {
    const dismissHandler = component.find({testID: 'dismissPlane'});
    dismissHandler.simulate('press');
    expect(props.onDismiss).toHaveBeenCalledTimes(1);
  });
  it('should render children', () => {
    const title = component.find({children: testLabel});
    expect(title).toHaveLength(1);
  });
});
