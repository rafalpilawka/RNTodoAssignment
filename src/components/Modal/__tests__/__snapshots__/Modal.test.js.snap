// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal should match to snapshot 1`] = `
<Modal
  animationType="slide"
  hardwareAccelerated={false}
  onRequestClose={[MockFunction]}
  supportedOrientations={
    Array [
      "portrait",
    ]
  }
  transparent={true}
  visible={true}
>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "rgba(0,0,0,0.5)",
        "flex": 1,
        "justifyContent": "flex-end",
        "opacity": 1,
      }
    }
    testID="dismissPlane"
  >
    <View
      onLayout={[Function]}
      style={
        Object {
          "paddingBottom": 0,
        }
      }
    >
      <RCTSafeAreaView
        emulateUnlessSupported={true}
        style={
          Array [
            Object {
              "backgroundColor": Object {
                "semantic": Array [
                  "systemBlue",
                ],
              },
              "justifyContent": "flex-end",
            },
          ]
        }
      >
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": Object {
                "semantic": Array [
                  "systemBlue",
                ],
              },
              "height": 400,
              "justifyContent": "flex-start",
              "opacity": 1,
              "paddingBottom": 50,
              "paddingHorizontal": "5%",
            }
          }
        />
      </RCTSafeAreaView>
    </View>
  </View>
</Modal>
`;
